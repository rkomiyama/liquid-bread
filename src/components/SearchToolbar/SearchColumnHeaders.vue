<template>
  <v-list dense>
    <v-subheader>Column headers</v-subheader>
    <v-list-tile v-for="(header, i) in headers" :key="i">
      <v-list-tile-action>
        <v-checkbox v-model="checked[i]"></v-checkbox>
      </v-list-tile-action>
      <v-list-tile-content>
        <v-list-tile-title>{{ header.text }}</v-list-tile-title>
      </v-list-tile-content>
    </v-list-tile>
  </v-list>
</template>

<script>
export default {
  name: "SearchColumnHeaders",
  data() {
    return {
      checked: []
    };
  },
  created() {
    this.checked = this.passedChecks;
  },
  watch: {
    checked(checked) {
      if (checked && checked.length) {
        this.$emit("change:checks", checked);
      }
    }
  },
  props: {
    headers: {
      type: Array,
      default: () => []
    },
    passedChecks: {
      type: Array,
      default: () => []
    }
  }
};
</script>

<style lang="scss" scoped>
</style>
