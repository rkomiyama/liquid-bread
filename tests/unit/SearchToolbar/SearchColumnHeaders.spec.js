import Vue from "vue";
import Vuetify from "vuetify";
import { shallowMount } from "@vue/test-utils";
import { createRenderer } from "vue-server-renderer";
import SearchColumnHeaders from "@/components/SearchToolbar/SearchColumnHeaders.vue";
import searchColumnHeaders from "#/mockData/searchColumnHeaders.json";

Vue.use(Vuetify);

let wrapper;
let checks = Array(searchColumnHeaders.length).fill(false);

describe("SearchColumnHeaders.vue", () => {
  beforeEach(() => {
    wrapper = shallowMount(SearchColumnHeaders, {
      propsData: {
        headers: searchColumnHeaders,
        passedChecks: checks
      }
    });
  });

  it("handles empty headers", () => {
    wrapper = shallowMount(SearchColumnHeaders);
    expect(wrapper.vm.headers).toBeUndefined();
  });

  it("emits change:checks", () => {
    wrapper = shallowMount(SearchColumnHeaders);
    expect(wrapper.emitted("change:checks")).toBeFalsy();
    wrapper.setData({ checked: [true] });
    expect(wrapper.emitted("change:checks")).toBeTruthy();
  });

  it("SearchColumnHeaders has same HTML structure", () => {
    const renderer = createRenderer();
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});
