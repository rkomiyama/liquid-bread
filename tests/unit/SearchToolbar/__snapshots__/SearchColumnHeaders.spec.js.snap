// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchColumnHeaders.vue SearchColumnHeaders has same HTML structure 1`] = `
<v-list-stub dense="true">
  <v-subheader-stub>Column headers</v-subheader-stub>
  <v-list-tile-stub activeClass="primary--text">
    <v-list-tile-action-stub>
      <v-checkbox-stub color="accent" errorCount="1" errorMessages="" messages="" rules="" successMessages="" backgroundColor="" ripple="true" valueComparator="function deepEqual(a, b) {
    if (a === b) return true;
    if (a instanceof Date &amp;&amp; b instanceof Date) {
        // If the values are Date, they were convert to timestamp with getTime and compare it
        if (a.getTime() !== b.getTime()) return false;
    }
    if (a !== Object(a) || b !== Object(b)) {
        // If the values aren't objects, they were already checked for equality
        return false;
    }
    var props = Object.keys(a);
    if (props.length !== Object.keys(b).length) {
        // Different number of props, don't bother to check
        return false;
    }
    return props.every(function (p) {
        return deepEqual(a[p], b[p]);
    });
}" indeterminateIcon="$vuetify.icons.checkboxIndeterminate" onIcon="$vuetify.icons.checkboxOn" offIcon="$vuetify.icons.checkboxOff"></v-checkbox-stub>
    </v-list-tile-action-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>ID</v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </v-list-tile-stub>
  <v-list-tile-stub activeClass="primary--text">
    <v-list-tile-action-stub>
      <v-checkbox-stub color="accent" errorCount="1" errorMessages="" messages="" rules="" successMessages="" backgroundColor="" ripple="true" valueComparator="function deepEqual(a, b) {
    if (a === b) return true;
    if (a instanceof Date &amp;&amp; b instanceof Date) {
        // If the values are Date, they were convert to timestamp with getTime and compare it
        if (a.getTime() !== b.getTime()) return false;
    }
    if (a !== Object(a) || b !== Object(b)) {
        // If the values aren't objects, they were already checked for equality
        return false;
    }
    var props = Object.keys(a);
    if (props.length !== Object.keys(b).length) {
        // Different number of props, don't bother to check
        return false;
    }
    return props.every(function (p) {
        return deepEqual(a[p], b[p]);
    });
}" indeterminateIcon="$vuetify.icons.checkboxIndeterminate" onIcon="$vuetify.icons.checkboxOn" offIcon="$vuetify.icons.checkboxOff"></v-checkbox-stub>
    </v-list-tile-action-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>Name</v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </v-list-tile-stub>
  <v-list-tile-stub activeClass="primary--text">
    <v-list-tile-action-stub>
      <v-checkbox-stub color="accent" errorCount="1" errorMessages="" messages="" rules="" successMessages="" backgroundColor="" ripple="true" valueComparator="function deepEqual(a, b) {
    if (a === b) return true;
    if (a instanceof Date &amp;&amp; b instanceof Date) {
        // If the values are Date, they were convert to timestamp with getTime and compare it
        if (a.getTime() !== b.getTime()) return false;
    }
    if (a !== Object(a) || b !== Object(b)) {
        // If the values aren't objects, they were already checked for equality
        return false;
    }
    var props = Object.keys(a);
    if (props.length !== Object.keys(b).length) {
        // Different number of props, don't bother to check
        return false;
    }
    return props.every(function (p) {
        return deepEqual(a[p], b[p]);
    });
}" indeterminateIcon="$vuetify.icons.checkboxIndeterminate" onIcon="$vuetify.icons.checkboxOn" offIcon="$vuetify.icons.checkboxOff"></v-checkbox-stub>
    </v-list-tile-action-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>ABV</v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </v-list-tile-stub>
  <v-list-tile-stub activeClass="primary--text">
    <v-list-tile-action-stub>
      <v-checkbox-stub color="accent" errorCount="1" errorMessages="" messages="" rules="" successMessages="" backgroundColor="" ripple="true" valueComparator="function deepEqual(a, b) {
    if (a === b) return true;
    if (a instanceof Date &amp;&amp; b instanceof Date) {
        // If the values are Date, they were convert to timestamp with getTime and compare it
        if (a.getTime() !== b.getTime()) return false;
    }
    if (a !== Object(a) || b !== Object(b)) {
        // If the values aren't objects, they were already checked for equality
        return false;
    }
    var props = Object.keys(a);
    if (props.length !== Object.keys(b).length) {
        // Different number of props, don't bother to check
        return false;
    }
    return props.every(function (p) {
        return deepEqual(a[p], b[p]);
    });
}" indeterminateIcon="$vuetify.icons.checkboxIndeterminate" onIcon="$vuetify.icons.checkboxOn" offIcon="$vuetify.icons.checkboxOff"></v-checkbox-stub>
    </v-list-tile-action-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>IBU</v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </v-list-tile-stub>
  <v-list-tile-stub activeClass="primary--text">
    <v-list-tile-action-stub>
      <v-checkbox-stub color="accent" errorCount="1" errorMessages="" messages="" rules="" successMessages="" backgroundColor="" ripple="true" valueComparator="function deepEqual(a, b) {
    if (a === b) return true;
    if (a instanceof Date &amp;&amp; b instanceof Date) {
        // If the values are Date, they were convert to timestamp with getTime and compare it
        if (a.getTime() !== b.getTime()) return false;
    }
    if (a !== Object(a) || b !== Object(b)) {
        // If the values aren't objects, they were already checked for equality
        return false;
    }
    var props = Object.keys(a);
    if (props.length !== Object.keys(b).length) {
        // Different number of props, don't bother to check
        return false;
    }
    return props.every(function (p) {
        return deepEqual(a[p], b[p]);
    });
}" indeterminateIcon="$vuetify.icons.checkboxIndeterminate" onIcon="$vuetify.icons.checkboxOn" offIcon="$vuetify.icons.checkboxOff"></v-checkbox-stub>
    </v-list-tile-action-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>EBC</v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </v-list-tile-stub>
  <v-list-tile-stub activeClass="primary--text">
    <v-list-tile-action-stub>
      <v-checkbox-stub color="accent" errorCount="1" errorMessages="" messages="" rules="" successMessages="" backgroundColor="" ripple="true" valueComparator="function deepEqual(a, b) {
    if (a === b) return true;
    if (a instanceof Date &amp;&amp; b instanceof Date) {
        // If the values are Date, they were convert to timestamp with getTime and compare it
        if (a.getTime() !== b.getTime()) return false;
    }
    if (a !== Object(a) || b !== Object(b)) {
        // If the values aren't objects, they were already checked for equality
        return false;
    }
    var props = Object.keys(a);
    if (props.length !== Object.keys(b).length) {
        // Different number of props, don't bother to check
        return false;
    }
    return props.every(function (p) {
        return deepEqual(a[p], b[p]);
    });
}" indeterminateIcon="$vuetify.icons.checkboxIndeterminate" onIcon="$vuetify.icons.checkboxOn" offIcon="$vuetify.icons.checkboxOff"></v-checkbox-stub>
    </v-list-tile-action-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>SRM</v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </v-list-tile-stub>
  <v-list-tile-stub activeClass="primary--text">
    <v-list-tile-action-stub>
      <v-checkbox-stub color="accent" errorCount="1" errorMessages="" messages="" rules="" successMessages="" backgroundColor="" ripple="true" valueComparator="function deepEqual(a, b) {
    if (a === b) return true;
    if (a instanceof Date &amp;&amp; b instanceof Date) {
        // If the values are Date, they were convert to timestamp with getTime and compare it
        if (a.getTime() !== b.getTime()) return false;
    }
    if (a !== Object(a) || b !== Object(b)) {
        // If the values aren't objects, they were already checked for equality
        return false;
    }
    var props = Object.keys(a);
    if (props.length !== Object.keys(b).length) {
        // Different number of props, don't bother to check
        return false;
    }
    return props.every(function (p) {
        return deepEqual(a[p], b[p]);
    });
}" indeterminateIcon="$vuetify.icons.checkboxIndeterminate" onIcon="$vuetify.icons.checkboxOn" offIcon="$vuetify.icons.checkboxOff"></v-checkbox-stub>
    </v-list-tile-action-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>First brewed</v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </v-list-tile-stub>
  <v-list-tile-stub activeClass="primary--text">
    <v-list-tile-action-stub>
      <v-checkbox-stub color="accent" errorCount="1" errorMessages="" messages="" rules="" successMessages="" backgroundColor="" ripple="true" valueComparator="function deepEqual(a, b) {
    if (a === b) return true;
    if (a instanceof Date &amp;&amp; b instanceof Date) {
        // If the values are Date, they were convert to timestamp with getTime and compare it
        if (a.getTime() !== b.getTime()) return false;
    }
    if (a !== Object(a) || b !== Object(b)) {
        // If the values aren't objects, they were already checked for equality
        return false;
    }
    var props = Object.keys(a);
    if (props.length !== Object.keys(b).length) {
        // Different number of props, don't bother to check
        return false;
    }
    return props.every(function (p) {
        return deepEqual(a[p], b[p]);
    });
}" indeterminateIcon="$vuetify.icons.checkboxIndeterminate" onIcon="$vuetify.icons.checkboxOn" offIcon="$vuetify.icons.checkboxOff"></v-checkbox-stub>
    </v-list-tile-action-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>Volume</v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </v-list-tile-stub>
  <v-list-tile-stub activeClass="primary--text">
    <v-list-tile-action-stub>
      <v-checkbox-stub color="accent" errorCount="1" errorMessages="" messages="" rules="" successMessages="" backgroundColor="" ripple="true" valueComparator="function deepEqual(a, b) {
    if (a === b) return true;
    if (a instanceof Date &amp;&amp; b instanceof Date) {
        // If the values are Date, they were convert to timestamp with getTime and compare it
        if (a.getTime() !== b.getTime()) return false;
    }
    if (a !== Object(a) || b !== Object(b)) {
        // If the values aren't objects, they were already checked for equality
        return false;
    }
    var props = Object.keys(a);
    if (props.length !== Object.keys(b).length) {
        // Different number of props, don't bother to check
        return false;
    }
    return props.every(function (p) {
        return deepEqual(a[p], b[p]);
    });
}" indeterminateIcon="$vuetify.icons.checkboxIndeterminate" onIcon="$vuetify.icons.checkboxOn" offIcon="$vuetify.icons.checkboxOff"></v-checkbox-stub>
    </v-list-tile-action-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>Atten. level</v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </v-list-tile-stub>
  <v-list-tile-stub activeClass="primary--text">
    <v-list-tile-action-stub>
      <v-checkbox-stub color="accent" errorCount="1" errorMessages="" messages="" rules="" successMessages="" backgroundColor="" ripple="true" valueComparator="function deepEqual(a, b) {
    if (a === b) return true;
    if (a instanceof Date &amp;&amp; b instanceof Date) {
        // If the values are Date, they were convert to timestamp with getTime and compare it
        if (a.getTime() !== b.getTime()) return false;
    }
    if (a !== Object(a) || b !== Object(b)) {
        // If the values aren't objects, they were already checked for equality
        return false;
    }
    var props = Object.keys(a);
    if (props.length !== Object.keys(b).length) {
        // Different number of props, don't bother to check
        return false;
    }
    return props.every(function (p) {
        return deepEqual(a[p], b[p]);
    });
}" indeterminateIcon="$vuetify.icons.checkboxIndeterminate" onIcon="$vuetify.icons.checkboxOn" offIcon="$vuetify.icons.checkboxOff"></v-checkbox-stub>
    </v-list-tile-action-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>pH</v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </v-list-tile-stub>
  <v-list-tile-stub activeClass="primary--text">
    <v-list-tile-action-stub>
      <v-checkbox-stub color="accent" errorCount="1" errorMessages="" messages="" rules="" successMessages="" backgroundColor="" ripple="true" valueComparator="function deepEqual(a, b) {
    if (a === b) return true;
    if (a instanceof Date &amp;&amp; b instanceof Date) {
        // If the values are Date, they were convert to timestamp with getTime and compare it
        if (a.getTime() !== b.getTime()) return false;
    }
    if (a !== Object(a) || b !== Object(b)) {
        // If the values aren't objects, they were already checked for equality
        return false;
    }
    var props = Object.keys(a);
    if (props.length !== Object.keys(b).length) {
        // Different number of props, don't bother to check
        return false;
    }
    return props.every(function (p) {
        return deepEqual(a[p], b[p]);
    });
}" indeterminateIcon="$vuetify.icons.checkboxIndeterminate" onIcon="$vuetify.icons.checkboxOn" offIcon="$vuetify.icons.checkboxOff"></v-checkbox-stub>
    </v-list-tile-action-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>Boil volume</v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </v-list-tile-stub>
  <v-list-tile-stub activeClass="primary--text">
    <v-list-tile-action-stub>
      <v-checkbox-stub color="accent" errorCount="1" errorMessages="" messages="" rules="" successMessages="" backgroundColor="" ripple="true" valueComparator="function deepEqual(a, b) {
    if (a === b) return true;
    if (a instanceof Date &amp;&amp; b instanceof Date) {
        // If the values are Date, they were convert to timestamp with getTime and compare it
        if (a.getTime() !== b.getTime()) return false;
    }
    if (a !== Object(a) || b !== Object(b)) {
        // If the values aren't objects, they were already checked for equality
        return false;
    }
    var props = Object.keys(a);
    if (props.length !== Object.keys(b).length) {
        // Different number of props, don't bother to check
        return false;
    }
    return props.every(function (p) {
        return deepEqual(a[p], b[p]);
    });
}" indeterminateIcon="$vuetify.icons.checkboxIndeterminate" onIcon="$vuetify.icons.checkboxOn" offIcon="$vuetify.icons.checkboxOff"></v-checkbox-stub>
    </v-list-tile-action-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>Target OG</v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </v-list-tile-stub>
  <v-list-tile-stub activeClass="primary--text">
    <v-list-tile-action-stub>
      <v-checkbox-stub color="accent" errorCount="1" errorMessages="" messages="" rules="" successMessages="" backgroundColor="" ripple="true" valueComparator="function deepEqual(a, b) {
    if (a === b) return true;
    if (a instanceof Date &amp;&amp; b instanceof Date) {
        // If the values are Date, they were convert to timestamp with getTime and compare it
        if (a.getTime() !== b.getTime()) return false;
    }
    if (a !== Object(a) || b !== Object(b)) {
        // If the values aren't objects, they were already checked for equality
        return false;
    }
    var props = Object.keys(a);
    if (props.length !== Object.keys(b).length) {
        // Different number of props, don't bother to check
        return false;
    }
    return props.every(function (p) {
        return deepEqual(a[p], b[p]);
    });
}" indeterminateIcon="$vuetify.icons.checkboxIndeterminate" onIcon="$vuetify.icons.checkboxOn" offIcon="$vuetify.icons.checkboxOff"></v-checkbox-stub>
    </v-list-tile-action-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>Target FG</v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </v-list-tile-stub>
</v-list-stub>
`;
